fieldservice/models/fsm_order.py

```diff
@@ -183,10 +183,12 @@ def _onchange_location_id_customer(self):
 
     @api.model
     def _read_group_stage_ids(self, stages, domain, order):
-        stage_ids = self.env['fsm.stage'].\
-            search([('stage_type', '=', 'order'),
-                    ('company_id', '=', self.env.user.company_id.id)])
-        return stage_ids
+        search_domain = [('stage_type', '=', 'order')]
+        if self.env.context.get('default_team_id'):
+            search_domain = [
+                '&', ('team_ids', 'in', self.env.context['default_team_id'])
+            ] + search_domain
+        return stages.search(search_domain, order=order)
```

Comment:
Can we have stages with no team_id ?

Reply:
If you aren't using teams in fieldservice, everything will be set to the default team and team fields are hidden.  So all stages would have a team.  team_ids is not a required field on stage, so maybe it should be? It's a required field on the order

Reply:
Ok then.